pkg/util/goroutinemap/exponentialbackoff/exponential_backoff.go

```diff
@@ -29,20 +29,15 @@ const (
 	// the same target (if exponentialBackOffOnError is enabled). Each
 	// successive error results in a wait 2x times the previous.
 	initialDurationBeforeRetry time.Duration = 500 * time.Millisecond
-
-	// maxDurationBeforeRetry is the maximum amount of time that
-	// durationBeforeRetry will grow to due to exponential backoff.
-	// Value is slightly offset from 2 minutes to make timeouts due to this
-	// constant recognizable.
-	maxDurationBeforeRetry time.Duration = 2*time.Minute + 2*time.Second
 )
 
 // ExponentialBackoff contains the last occurrence of an error and the duration
 // that retries are not permitted.
 type ExponentialBackoff struct {
-	lastError           error
-	lastErrorTime       time.Time
-	durationBeforeRetry time.Duration
+	lastError              error
+	lastErrorTime          time.Time
+	durationBeforeRetry    time.Duration
+	MaxDurationBeforeRetry time.Duration
```

Comment:
all the other fields are unexported, I'd expect this one to be as well. if you need a constructor to create a new ExponentialBackoff with a specific maxDuration, create a constructor (e.g. `NewWithMaxDurationBeforeRetry`) and error if <= 0 is passed